### ISSUE TITLE
[Accessibility]: Search results misc keyboard bugs

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34723]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

The following problems exist for keyboard navigation on the result page:

### On the header:

![image](https://user-images.githubusercontent.com/206864/166072481-1410862c-d63b-483a-82db-09728d0d5ece.png)
- [ ] Cannot focus (i) icon next to results count with keyboard to see number of repos
- [ ] "Create context", "Create insight", "Monitor" and "Save Search" buttons cannot be activated with spacebar
- [ ] Clicking on "save search" and then clicking "save query" on the dialog navigates to the save search page, the description box should be focused by default

###On search results items:
![image](https://user-images.githubusercontent.com/206864/166072440-9e7654f1-5293-48c1-bbc9-143532d9605e.png)

- [ ] Last sync time icon cannot be focused with keyboard to see the last synced time

On the no results page:
![image](https://user-images.githubusercontent.com/206864/166072406-3c247091-70f9-429a-a1fd-4f108356491e.png)

- [ ] "Run search" link in examples cannot be focused with the keyboard
- [ ] Odd focus ring around no results page examples ![image](https://user-images.githubusercontent.com/206864/166072574-0cd459ac-1f83-4005-9c5e-206fb5bdcbc3.png)
- [ ] Should not be able to focus toggles on no results page examples

### Expected behavior

All issues should be resolved

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
